src/compiler/program.ts

```diff
@@ -1334,7 +1361,10 @@ namespace ts {
                 getNewLine: () => host.getNewLine(),
                 getSourceFile: program.getSourceFile,
                 getSourceFileByPath: program.getSourceFileByPath,
-                getSourceFiles: program.getSourceFiles,
+                getSourceFiles: () => filter(program.getSourceFiles(),
+                    // Remove JavaScript files found by searching node_modules from the source files to emit
+                    sourceFile => !lookUp(jsFilesFoundSearchingNodeModules, sourceFile.path)

```

Comment:
do not think we should do this. the file is part of the program. getSourceFle, and getSourceFile, should give an accurate representation of the files in the program. you could get an error in this file, or get a definition in this file, and using the file name you should be able to get to the SourceFile object using `getSourceFile` and you should also be able to find it if you enumerate the `getSourceFiles`.

I would say create a new method, `Program.jsFilesFoundSearchingNodeModules(path)` that the emitter checks and skips emitting if true.